src/components/widgets/support-case-widget.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Table,
+  TableVariant,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from '@patternfly/react-table';
+import { Button } from '@patternfly/react-core/dist/dynamic/components/Button';
+import { EmptyState } from '@patternfly/react-core/dist/dynamic/components/EmptyState';
+import { EmptyStateBody } from '@patternfly/react-core/dist/dynamic/components/EmptyState';
+import { EmptyStateIcon } from '@patternfly/react-core/dist/dynamic/components/EmptyState';
+import { EmptyStateVariant } from '@patternfly/react-core/dist/dynamic/components/EmptyState';
+import { Stack } from '@patternfly/react-core/dist/dynamic/layouts/Stack';
+import { StackItem } from '@patternfly/react-core/dist/dynamic/layouts/Stack';
+import { Title } from '@patternfly/react-core/dist/dynamic/components/Title';
+import ExternalLinkAltIcon from '@patternfly/react-icons/dist/dynamic/icons/external-link-alt-icon';
+import { Label } from '@patternfly/react-core/dist/dynamic/components/Label';
+import HeadsetIcon from '@patternfly/react-icons/dist/dynamic/icons/headset-icon';
+import useCurrentUser from '../useCurrentUser';
+
+export type Case = {
+  id: string;
+  caseNumber: string;
+  summary: string;
+  lastModifiedById: string;
+  severity: string;
+  status: string;
+  isLoading: boolean;
+};
+
+const SupportCaseWidget: React.FunctionComponent = () => {
+  const [cases, setCases] = useState<Case[]>([]);
+  const MAX_ROWS = 5;
+  const { currentUser } = useCurrentUser();
+
+  const fetchSupportCases = async () => {
+    // eslint-disable-next-line rulesdir/no-chrome-api-call-from-window
+    const token = await window.insights.chrome.auth.getToken();
+    const url = 'https://api.access.stage.redhat.com/support/v1/cases/filter';
+    const options = {
+      method: 'POST',
+      headers: {
+        Authorization: `Bearer ${token}`,
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+      body: JSON.stringify({
+        createdBySSOName: `${currentUser?.username}`,
```

Comment:
this `${currentUser.username}` comes back as undefined, If I check the currentUser it comes back with the right data but it's like it can't reach it here for some reason --  @florkbr any suggestions here?

Reply:
Maybe the request is just fired before the user data is retrieved, maybe you could use `await chrome.auth.getUser();` similarly to the token retrieval